x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/crawler/crawler_domain_detail/extraction_rules/extraction_rules_logic.tsx

```diff
@@ -0,0 +1,329 @@
+  deleteFieldModalVisible: boolean;
+  deleteModalVisible: boolean;
+  deleteStatus: Status;
+  domain: CrawlerDomainDetailValues['domain'];
+  domainExtractionRules: ExtractionRule[] | null;
+  domainId: string;
+  editingExtractionRule: boolean;
+  extractionRuleToDelete: ExtractionRule | null;
+  extractionRuleToEdit: ExtractionRule | null;
+  extractionRuleToEditIsNew: boolean;
+  extractionRules: ExtractionRule[];
+  fieldRuleFlyoutVisible: boolean;
+  fieldRuleToDelete: { extractionRuleId?: string; fieldRuleIndex?: number };
+  fieldRuleToEdit: ExtractionRuleFieldRule | null;
+  fieldRuleToEditIndex: number | null;
+  fieldRuleToEditIsNew: boolean;
+  indexName: string;
+  isLoading: boolean;
+  isLoadingUpdate: boolean;
+  jsonValidationError: boolean;
+  updateStatus: Status;
+  updatedExtractionRules: ExtractionRule[] | null;
+}
+
+export const ExtractionRulesLogic = kea<
+  MakeLogicType<ExtractionRulesValues, ExtractionRulesActions>
+>({
+  actions: {
+    cancelEditExtractionRule: true,
+    closeEditRuleFlyout: true,
+    deleteExtractionRule: true,
+    deleteFieldRule: true,
+    editExtractionRule: (extractionRule) => ({ extractionRule }),
+    editNewExtractionRule: true,
+    hideDeleteFieldModal: true,
+    hideDeleteModal: true,
+    openEditRuleFlyout: ({ fieldRule, isNewRule }) => ({
+      fieldRule,
+      isNewRule,
+    }),
+    saveExtractionRule: (extractionRule: ExtractionRuleBase) => ({ extractionRule }),
+    showDeleteFieldModal: ({ fieldRuleIndex, extractionRuleId }) => ({
+      extractionRuleId,
+      fieldRuleIndex,
+    }),
+    showDeleteModal: (extractionRule: ExtractionRule) => ({ extractionRule }),
+  },
+  connect: {
+    actions: [
+      AddExtractionRuleApiLogic,
+      ['makeRequest as addExtractionRule', 'apiSuccess as addExtractionRuleSuccess'],
+      CrawlerDomainDetailLogic,
+      ['receiveDomainData'],
+      DeleteExtractionRuleApiLogic,
+      ['makeRequest as deleteExtractionRuleRequest', 'apiSuccess as deleteExtractionRuleSuccess'],
+      FetchExtractionRulesApiLogic,
+      ['makeRequest as fetchExtractionRules', 'apiSuccess as fetchExtractionRulesSuccess'],
+      UpdateExtractionRuleApiLogic,
+      ['makeRequest as updateExtractionRule', 'apiSuccess as updateExtractionRuleSuccess'],
+    ],
+    values: [
+      AddExtractionRuleApiLogic,
+      ['status as addStatus'],
+      CrawlerDomainDetailLogic,
+      ['domain', 'domainId', 'extractionRules as domainExtractionRules', 'getLoading as isLoading'],
+      DeleteExtractionRuleApiLogic,
+      ['status as deleteStatus'],
+      IndexNameLogic,
+      ['indexName'],
+      UpdateExtractionRuleApiLogic,
+      ['status as updateStatus'],
+    ],
+  },
+  events: ({ actions, values }) => ({
+    beforeUnmount: () => {
+      // make sure we feed back our updated extraction rules data to the domain
+      if (values.domain) {
+        actions.receiveDomainData({ ...values.domain, extractionRules: values.extractionRules });
```

Comment:
This isn't so much performance related as consistency related (switching tabs made the outdated state reappear because the logic gets unmounted), but I think you're right that refetching probably makes more sense.